proto/data.go

```diff
@@ -316,6 +317,69 @@ func (v *Value) SetProto(msg gogoproto.Message) error {
 	return nil
 }
 
+// SetTime encodes the specified time value into the bytes field of the
+// receiver and sets the tag.
+func (v *Value) SetTime(t time.Time) error {
+	var err error
+	if v.Bytes, err = t.MarshalBinary(); err != nil {
```

Comment:
One option is to panic if the offset is not UTC. Another option is to just
store it and not bother. Timezone is only used for parsing an input
timestamp and presenting it out to the user based on their preference.

On Tue, Sep 22, 2015 at 11:16 AM Tamir Duberstein <notifications@github.com>
wrote:

> In proto/data.go
> <https://github.com/cockroachdb/cockroach/pull/2603#discussion_r40096903>:
>
> > @@ -316,6 +317,69 @@ func (v *Value) SetProto(msg gogoproto.Message) error {
> >  	return nil
> >  }
> >
> > +// SetTime encodes the specified time value into the bytes field of the
> > +// receiver and sets the tag.
> > +func (v *Value) SetTime(t time.Time) error {
> > +	var err error
> > +	if v.Bytes, err = t.MarshalBinary(); err != nil {
>
> Interesting point. We'll eventually need to either:
>
>    - plumb a custom time zone into this method (see discussion in #2565
>    <https://github.com/cockroachdb/cockroach/pull/2565> about #2327
>    <https://github.com/cockroachdb/cockroach/issues/2327>)
>    - set the custom time zone in the caller, which means we should *not*
>    muck with the time zone in this method.
>
> On the other hand, assuming we handle time conversions correctly
> everywhere else, the offset we store doesn't actually matter, and we should
> just pick the fastest option (which is probably the current code).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2603/files#r40096903>.
>